src/world/StatusEffect/StatusEffect.cpp

```diff
@@ -153,3 +254,75 @@ const std::string& Sapphire::StatusEffect::StatusEffect::getName() const
 {
   return m_name;
 }
+
+const Sapphire::World::Action::StatusEffectEntry& Sapphire::StatusEffect::StatusEffect::getEffectEntry() const
+{
+  return m_effectEntry;
+}
+
+void Sapphire::StatusEffect::StatusEffect::replaceEffectEntry( Sapphire::World::Action::StatusEffectEntry entryOverride )
+{
+  m_effectEntry = entryOverride;
+}
+
+void Sapphire::StatusEffect::StatusEffect::onBeforeActionStart( Sapphire::World::Action::ActionPtr action )
+{
+  // todo: add script function for this if needed
+  //auto pScriptMgr = m_pFw->get< Scripting::ScriptMgr >();
+  //pScriptMgr->onBeforeActionStart( m_targetActor, *this );
+
+  switch( static_cast< Common::StatusEffectType >( m_effectEntry.effectType ) )
+  {
+    case Common::StatusEffectType::InstantCast:
+    {
+      if( !action->hasCastTime() )
+        return;
+      // value1: remaining uses
+      // value2-4: affected action ids, or all actions if value2 is 0
+      if( m_effectEntry.effectValue2 != 0 )
+      {
+        if( action->getId() != m_effectEntry.effectValue2 &&
```

Comment:
this effectValue stuff is incredibly obscure, I don't like this as a solution but I don't think it'd be much better even if it were a union - see review message

Reply:
I agree with it however I don't think there's any better way than a union, and also easy to generate.